compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -202,7 +202,7 @@ private boolean isDistinctSymbol(DiagnosticPos pos, BSymbol symbol, BSymbol foun
         }
 
         // Type names should be unique and cannot be shadowed
-        if ((foundSym.tag & SymTag.TYPE) == SymTag.TYPE) {
+        if ((foundSym.tag & SymTag.TYPE) == SymTag.TYPE && !((foundSym.tag & SymTag.CONSTANT) == SymTag.CONSTANT)) {
```

Comment:
I feel like if we move the complex logic to a function it would enhance readability. WDYT?

Reply:
It is not used anymore.